From a13b6c39677bb2a4edd7207f2cf579b7ebd7f771 Mon Sep 17 00:00:00 2001
From: Meng Dongyang <daniel.meng@rock-chips.com>
Date: Fri, 1 Jun 2018 11:17:01 +0800
Subject: [PATCH] usb: gadget: uac: add sample rate to attribute of uevent

Add sample rate to attribute of uevent when receive the command
of set interface.

Change-Id: I9cea17ebe57441209a99dec0b1dc279e274accc3
Signed-off-by: Meng Dongyang <daniel.meng@rock-chips.com>
---
 drivers/usb/gadget/function/f_audio_source.c | 12 +++++++++---
 1 file changed, 9 insertions(+), 3 deletions(-)

diff --git a/drivers/usb/gadget/function/f_audio_source.c b/drivers/usb/gadget/function/f_audio_source.c
index 573a938a5a3d..568cb897dcae 100644
--- a/drivers/usb/gadget/function/f_audio_source.c
+++ b/drivers/usb/gadget/function/f_audio_source.c
@@ -340,13 +340,19 @@ static inline struct audio_dev *func_to_audio(struct usb_function *f)
 static void audio_source_work(struct work_struct *data)
 {
 	struct audio_dev *audio = container_of(data, struct audio_dev, work);
-	char *set_interface[3]	= { "USB_STATE=SET_INTERFACE", NULL, NULL};
+	char buffer[64];
+	char *set_interface[4]	= { "USB_STATE=SET_INTERFACE", NULL, NULL,
+				    NULL };
 	char **uevent_envp = NULL;
 
 	if (audio->alt)
-		set_interface[1] = "1";
+		set_interface[1] = "STREAM_STATE=ON";
 	else
-		set_interface[1] = "0";
+		set_interface[1] = "STREAM_STATE=OFF";
+
+	sprintf(buffer, "SAMPLE_RATE=%lld", audio->sample_rate);
+		set_interface[2] = buffer;
+
 	uevent_envp = set_interface;
 
 	if (uevent_envp) {
-- 
2.35.3

